src/RageFileManager.cpp

```diff
@@ -227,42 +234,41 @@ bool RageFileManager::Unzip(const std::string &zipPath, std::string targetPath,
 	return success;
 }
 
-/* Wait for the given driver to become unreferenced, and remove it from the list
- * to get exclusive access to it.  Returns false if the driver is no longer available
- * (somebody else got it first). */
-#if 0
-static bool GrabDriver( RageFileDriver *pDriver )
+static void NormalizePath( RString &sPath )
 {
-	g_Mutex->Lock();
+	FixSlashesInPlace( sPath );
+	CollapsePath( sPath, true );
+	if (sPath.size() == 0)
+	{
+		sPath = '/';
+	}
+	else if (sPath[0] != '/')
+	{
+		sPath = '/' + sPath;
+	}
+}
+
+bool RageFileManager::IsPathProtected(const std::string& path)
+{
+	static const std::string protectedPaths[] = {
+		SpecialFiles::PREFERENCES_INI_PATH,
```

Comment:
I think we should also add STATIC_INI_PATH as those override preferences

Reply:
Also it's good practice to have static variables either be trivially destructible (which std::strings are not), or construct them on first use using a function.

https://isocpp.org/wiki/faq/ctors#static-init-order

Since this is a list of strings, I think having a function is better.

```c++
const std::vector<std::string>& GetProtectedPaths() {
  static const auto* kProtectedPaths = new std::vector<std::string>({
      SpecialFiles::PREFERENCES_INI_PATH
  });
  return *kProtectedPaths;
}

bool RageFileManager::IsPathProtected(const std::string& path) {
  const auto& protectedPaths = GetProtectedPaths();
  ..
}
```

Reply:
oh, thanks for pointing this out! I wasn't even thinking of this.